src/cdomains/floatDomain.ml

```diff
@@ -283,6 +295,48 @@ module FloatInterval = struct
 
   let ne = eval_int_binop eval_ne
 
+  let true_nonZero_IInt = IntDomain.IntDomTuple.of_excl_list IInt [(Big_int_Z.big_int_of_int 0)]
```

Comment:
there is even a constructor for `IntDomTuple.of_bool` which might be event better suited for this case. We could even use `IBool` ikind.

Reply:
I would not, because the return type is not an `_Bool` type (`IBool` in Cil), but an `int` type (`IInt` in Cil). 
I am not sure how it is implemented in the int domain, but in the C standard it is defined that for example `x < y` is either 1 or 0 (which would be a `_Bool`) [see C99 page 86, 6.5.8, 6], while the `isnan` and similar float comparison macros are defined to return a non zero value or zero [see C99 page 217, 7.12.3.4, 2]

Reply:
I though `IntDomTuple.of_bool` with `IInt` would create just what you did manually here, but thats not the case. So you are just right.